pyfr/integrators/dual/pseudo/pseudosteppers.py

```diff
@@ -239,42 +263,27 @@ def pseudo_stepper_nregs(self):
     def step(self, t):
         self.npseudosteps += 1
 
-        add, rhs = self._add, self._rhs_with_dts
-        errest = self.pseudo_stepper_has_lerrest
+        q, rhs = self._queue, self._rhs_with_dts
 
         rold = self._idxcurr
-
-        if errest:
-            r2, r1, rerr = set(self._pseudo_stepper_regidx) - {rold}
-        else:
-            r2, r1 = set(self._pseudo_stepper_regidx) - {rold}
-
-        # Copy the current solution
-        add(0.0, r1, 1.0, rold)
+        r1, r2, *rerr = set(self._pseudo_stepper_regidx) - {rold}
 
         # Evaluate the stages in the scheme
         for i in range(self._nstages):
             # Compute -∇·f
-            rhs(t, r2 if i > 0 else r1, r2)
-
-            self.localdtau(r2)
+            rhs(t, r2 if i > 0 else rold, r2)
 
-            if errest:
-                # Accumulate the error term in rerr
-                add(1.0 if i > 0 else 0.0, rerr, self.e[i], r2)
+            # Fetch the appropriate RK accumulation kernels
+            kerns = self._get_rkvdh2pseudo_kerns(i, r1, r2, rold, *rerr)
 
-            # Sum (special-casing the final stage)
-            if i < self._nstages - 1:
-                add(1.0, r1, self.a[i], r2)
-                add(self.b[i] - self.a[i], r2, 1.0, r1)
-            else:
-                add(1.0, r1, self.b[i], r2)
+            # Execute
+            q.enqueue_and_run(kerns)
 
             # Swap
             r1, r2 = r2, r1
 
         # Return
-        return (r2, rold, rerr) if errest else (r2, rold)
+        return (r2, rold, *rerr)
```

Comment:
Why is the return block now different?

Reply:
Behavior is the same actually. rerr sometimes is an empty list and if so returned tuple is (r2, rold,). when it is not an empty list then there is an additional element in the tuple and this matches the requirement of the controller.